includes/class-wc-payments-order-service.php

```diff
@@ -54,12 +59,26 @@ public function mark_payment_completed( $order, $intent_id, $intent_status, $cha
 			return;
 		}
 
-		$note = $this->generate_payment_success_note( $intent_id, $charge_id, $this->get_order_amount( $order ) );
+		$note = $this->generate_payment_success_note( $charge_id, $this->get_order_amount( $order ) );
 
 		if ( $this->order_note_exists( $order, $note ) ) {
 			return;
 		}
 
+		// Dispatch call here to update the order note with fee breakdown.
+		$action_scheduler_service = new WC_Payments_Action_Scheduler_Service( $this->api_client );
```

Comment:
I think we might avoid initializing `WC_Payments_Action_Scheduler_Service` here as it can create a side effect (see its `__construct` that includes `add_action` functions). 

IMO, there are two ways for avoiding this: 

(1) Create a static function in `WC_Payments::get_action_scheduler_service` (which is similar to multiple static functions like `WC_Payments::get_localization_service`, and use it here.


(2) Add `WC_Payments_Action_Scheduler_Service` as a dependency for this Order_Service class, and register it in WC_Payments  - [src](https://github.com/Automattic/woocommerce-payments/blob/72f6bb76ecd432f09b598e2485aaf012bcf9a6d7/includes/class-wc-payments.php#L303). This requires multiple changes in our tests too because the class signature has changed. But the good thing is that it will be helpful when we need to refactor our code towards DI. 

I think using approach (1) is reasonable here, though (2) is more favorable in the long term.

Reply:
Updated this to use approach 1.